pkg/provider/cloud/baremetal/provider.go

```diff
@@ -32,11 +35,36 @@ func NewCloudProvider() provider.CloudProvider {
 
 var _ provider.CloudProvider = &baremetal{}
 
-func (b *baremetal) DefaultCloudSpec(_ context.Context, _ *kubermaticv1.ClusterSpec) error {
+func (b *baremetal) DefaultCloudSpec(_ context.Context, spec *kubermaticv1.ClusterSpec) error {
+	if spec.Cloud.Baremetal == nil {
+		return errors.New("baremetal cloud provider spec is empty")
+	}
+
+	if spec.Cloud.Baremetal != nil && spec.Cloud.Baremetal.Tinkerbell == nil {
+		return errors.New("tinkerbell spec is empty! we must have a provisioner")
+	}
 	return nil
 }
 
-func (b *baremetal) ValidateCloudSpec(_ context.Context, _ kubermaticv1.CloudSpec) error {
+func (b *baremetal) ValidateCloudSpec(_ context.Context, spec kubermaticv1.CloudSpec) error {
+	if spec.Baremetal.Tinkerbell != nil {
+		kubeconfig := spec.Baremetal.Tinkerbell.Kubeconfig
+		if kubeconfig == "" {
+			return errors.New("tinkerbell kubeconfig is not found")
```

Comment:
"is not found" sounds like it's a bad filename. "is empty" is a better expression